lib/mllp/sender.ex

```diff
@@ -1,32 +1,24 @@
 defmodule MLLP.SenderContract do
   @type error_type :: :connect_failure | :send_error | :recv_error
-  @type error_reason :: :closed | :timeout | :no_socket | :inet.posix()
+  @type error_reason :: :closed | :timeout | :not_connected | :inet.posix()
 
   @type sender_error :: %{type: error_type(), reason: error_reason()}
 
-  @callback send_hl7_and_receive_ack(
+  @type send_options :: %{
+          optional(:expect_reply) => boolean(),
+          optional(:timeout) => non_neg_integer(),
+          optional(:wrap_message) => boolean(),
+          optional(:reply_timeout) => non_neg_integer() | :infinity
+        }
+
+  @callback send(
```

Comment:
I see we had some non_hl7 and raw sends, which I believe are not used normally. I haven't seen anyone requesting non_hl7 messages on MLLP. We do have use case to receive non_hl7 on MLLP, but with time, it may go away too.

@bryanhunter - Do you see any need of non_hl7 send support?